libraries/AP_HAL_ChibiOS/CANFDIface.cpp

```diff
@@ -568,7 +568,7 @@ bool CANIface::init(const uint32_t bitrate, const uint32_t fdbitrate, const Oper
     if (can_ifaces[self_index_] == nullptr) {
         can_ifaces[self_index_] = this;
 #if !defined(HAL_BOOTLOADER_BUILD)
-        hal.can[self_index_] = this;
+        const_cast <AP_HAL::HAL&> (hal).can[self_index_] = this;
```

Comment:
can we add a non-const method on hal, so we do:
  hal.non_const().can[1] = foo;
?

Reply:
I don’t think there’s a way to do that without invoking UB, a setter is a much better way. I will do that.

Reply:
A setter is not even needed if the CAN array is marked `mutable`. This is what I chose to do.